CoreScriptsRoot/Modules/BackpackScript3D.lua

```diff
@@ -39,29 +52,41 @@ local function UpdateLayout()
 	local borderSize = (ICON_SPACING - ICON_SIZE) / 2	
 	
 	local x = borderSize
+	local y = HEALTHBAR_SPACE
 	for _, tool in ipairs(ToolsList) do
 		local slot = Tools[tool]
 		if slot then
-			slot.icon.Position = UDim2.new(0, x, 0, 0)
+			slot.icon.Position = UDim2.new(0, x, 0, y)
 			x = x + ICON_SPACING
 		end
 	end
 	
 	width = #ToolsList * ICON_SPACING
-	height = ICON_SIZE
+	height = ICON_SIZE + HEALTHBAR_SPACE
 	
 	hopperbinGUI.CanvasSize = Vector2.new(width, height)
 	hopperbinPart.Size = Vector3.new(width / PIXELS_PER_STUD, height / PIXELS_PER_STUD, 1)	
+
+	healthbarBack.Position = UDim2.new(0.5, -HEALTHBAR_WIDTH / 2, 0, (HEALTHBAR_SPACE - HEALTHBAR_HEIGHT) / 2)
+	healthbarBack.Size = UDim2.new(0, HEALTHBAR_WIDTH, 0, HEALTHBAR_HEIGHT)
 	
 	verticalRange = math.atan(hopperbinPart.Size.Y / (2 * HOPPERBIN_OFFSET.Z)) + math.rad(30) * 2
 	horizontalRange = math.atan(hopperbinPart.Size.X / (2 * HOPPERBIN_OFFSET.Z)) - math.rad(20) * 2
 end
 
+local function UpdateHealth(humanoid)
+	local percent = humanoid.Health / humanoid.MaxHealth
+	healthbarFront.Size = UDim2.new(percent, 0, 1, 0)
```

Comment:
Alright, this is something to circle back with design (they might have some idea for what it looks like)